domain/model/service/modelservice.go

```diff
@@ -157,3 +212,23 @@ func (s *ModelService) GetStatus(ctx context.Context) (model.StatusInfo, error)
 		Since:  now,
 	}, nil
 }
+
+// GetEnvironProviderVersion retrieves the version of the environment provider associated with the model.
+//
+// Errors:
+// - [modelerrors.NotFound]: Returned if the model does not exist.
+func (s *ModelService) GetEnvironProviderVersion(ctx context.Context) (int, error) {
+	model, err := s.modelSt.GetModel(ctx)
+	if err != nil {
+		return 0, errors.Capture(err)
+	}
+
+	envProvider, err := s.environProviderGetter(model.CloudType)
+	if err != nil {
+		return 0, fmt.Errorf(
```

Comment:
Can we use the `Errorf` from `internal/errors` please.

Reply:
Noted with thanks